lib/AST/ASTImporter.cpp

```diff
@@ -4595,22 +4650,20 @@ Decl *ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
     }
   }
 
-  TemplateParameterList *Params = ImportTemplateParameterList(
-                                    D->getTemplateParameters());
+  TemplateParameterList *Params =
+      ImportTemplateParameterList(D->getTemplateParameters());
   if (!Params)
     return nullptr;
 
+  FunctionDecl *TemplatedFD =
```

Comment:
This code is moved down from above. It make more sense to try to import the templated function only after we made sure no such function already exist.